test/EFCore.PG.FunctionalTests/Query/HstoreQueryTest.cs

```diff
@@ -0,0 +1,281 @@
+    }
+
+    [Theory]
+    [MemberData(nameof(IsAsyncData))]
+    public async Task Dictionary_Item_equals_not_found(bool async)
+    {
+        var keyToTest = "key";
+        var valueToTest = "value2";
+        await AssertQuery(async, _ => _.Set<DictionaryEntity>().Where(s => s.Dictionary.ContainsKey(keyToTest) && s.Dictionary[keyToTest] == valueToTest), assertEmpty: true);
+        AssertSql(
+            """
+@__keyToTest_0='key'
+@__valueToTest_1='value2'
+
+SELECT s."Id", s."Dictionary", s."DictionaryContainerEntityId", s."ImmutableDictionary", s."NullableDictionary", s."NullableImmutableDictionary"
+FROM "SomeEntities" AS s
+WHERE s."Dictionary" ? @__keyToTest_0 AND s."Dictionary" -> 'key' = @__valueToTest_1
+""");
+    }
+
+    [Theory]
+    [MemberData(nameof(IsAsyncData))]
+    public async Task NullableDictionary_Item_equals(bool async)
+    {
+        var keyToTest = "key2";
+        var valueToTest = "value";
+        await AssertQuery(
+            async,
+            _ => _.Set<DictionaryEntity>().Where(
+                s => s.NullableDictionary != null
+                    && s.NullableDictionary.ContainsKey(keyToTest)
+                    && s.NullableDictionary[keyToTest] == valueToTest));
+        AssertSql(
+            """
+@__keyToTest_0='key2'
+@__valueToTest_1='value'
+
+SELECT s."Id", s."Dictionary", s."DictionaryContainerEntityId", s."ImmutableDictionary", s."NullableDictionary", s."NullableImmutableDictionary"
+FROM "SomeEntities" AS s
+WHERE s."NullableDictionary" IS NOT NULL AND s."NullableDictionary" ? @__keyToTest_0 AND s."NullableDictionary" -> 'key2' = @__valueToTest_1
+""");
+    }
+
+    [Theory]
+    [MemberData(nameof(IsAsyncData))]
+    public async Task ImmutableNullableDictionary_Item_equals(bool async)
+    {
+        var keyToTest = "key3";
+        var valueToTest = "value2";
+        await AssertQuery(
+            async,
+            _ => _.Set<DictionaryEntity>().Where(
+                s => s.NullableImmutableDictionary != null
+                    && s.NullableImmutableDictionary.ContainsKey(keyToTest)
+                    && s.NullableImmutableDictionary[keyToTest] == valueToTest));
+        AssertSql(
+            """
+@__keyToTest_0='key3'
+@__valueToTest_1='value2'
+
+SELECT s."Id", s."Dictionary", s."DictionaryContainerEntityId", s."ImmutableDictionary", s."NullableDictionary", s."NullableImmutableDictionary"
+FROM "SomeEntities" AS s
+WHERE s."NullableImmutableDictionary" IS NOT NULL AND s."NullableImmutableDictionary" ? @__keyToTest_0 AND s."NullableImmutableDictionary" -> @__keyToTest_0 = @__valueToTest_1
+""");
+    }
+
+    [Theory]
+    [MemberData(nameof(IsAsyncData))]
+    public async Task ImmutableDictionary_Item_equals(bool async)
+    {
+        var keyToTest = "key2";
+        var valueToTest = "value2";
+        await AssertQuery(async, _ => _.Set<DictionaryEntity>().Where(s => s.ImmutableDictionary.ContainsKey(keyToTest) && s.ImmutableDictionary[keyToTest] == valueToTest));
```

Comment:
You shouldn't need the ContainsKey() here - it's best for each test to test exactly one minimal thing. This probably fails in the cilent-side evaluation - AssertQuery accepts a second lambda to evaluate client-side, and there you can have the extra ContainsKey just to have the code work.